app/lib/features/chat/pages/chats_shell.dart

```diff
@@ -12,22 +14,33 @@ class ChatShell extends ConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final currentId = ref.watch(selectedChatIdProvider);
+    final isExpanded = ref.watch(hasExpandedPanel);
     return LayoutBuilder(
       builder: (context, constraints) {
         if (constraints.maxWidth >= sidebarMinWidth) {
+          WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+            ref.read(inSideBarProvider.notifier).update((state) => true);
+          });
```

Comment:
doing these kinda responsive things with providers always feels very weird too many. And using `addPostFrameCallback` even more ...  I feel like often times this call  has caused us quite some harm as it isn't actually reactive...

but let's do this for now ...

Reply:
The code is okay but not a pattern. It's very much go against the riverpod practices as it frequently changes the state based on ```constraints``` but ideally we should have some separate notifier which listens to it and do under the hood state change rather than doing inside ```build()```. This can be looked later but I'll add comment for improvement.